Extend arm-resource-patch rule with PATCH body schema diagnostics (#4381)
Azure/typespec-azure: website/src/content/docs/docs/libraries/azure-resource-manager/rules/arm-resource-patch.md

@azure-tools/typespec-azure-resource-manager/arm-resource-patch
```

Validate ARM PATCH operations. The request body of a PATCH must be a model with a subset of the resource properties. The PATCH body must not contain properties that do not exist on the resource. In addition, the rule validates that:

> - All properties in the PATCH request body are optional (PATCH supports partial updates), unless their source resource property has visibility `{Lifecycle.Read}` by itself (in which case they are filtered out of the request body by visibility transforms).

Reviewer: ```suggestion
- All properties in the PATCH request body are optional (PATCH supports partial updates) or read-only (not included in requests).
```

Reviewer: make this change in source so it appears here

- No PATCH request body property has a default value (a property absent from a PATCH request leaves the existing value unchanged; defaults are not applied).

> - Every PATCH request body property maps back to a resource property whose visibility either includes `Lifecycle.Update` (this includes default visibility, `@visibility(Lifecycle.Update)` alone, or any combination of `Lifecycle.Update` with other lifecycle modifiers) or is exactly `{Lifecycle.Read}` by itself. Visibilities like `@visibility(Lifecycle.Create)` or `@visibility(Lifecycle.Create, Lifecycle.Read)` are not allowed. The check is applied recursively to nested model and `Record<Model>` property types.

Reviewer: ```suggestion
- Every PATCH request body property maps back to a resource property that is updateable or readOnly.
```
